packages/next/src/server/app-render/app-render.tsx

```diff
@@ -733,13 +740,6 @@ async function renderToHTMLOrFlightImpl(
     requestId = require('next/dist/compiled/nanoid').nanoid()
   }
 
-  // During static generation we need to call the static generation bailout when reading searchParams
-  const providedSearchParams = isStaticGeneration
-    ? createSearchParamsBailoutProxy()
-    : query
-
-  const searchParamsProps = { searchParams: providedSearchParams }
```

Comment:
We move this as close to where it is used as possible. see `createDynamicallyTrackedSearchParams` for how this is done now